packages/react/src/AuthorizedContent.tsx

```diff
@@ -1,39 +1,56 @@
 import React, { FC, ReactNode } from 'react';
-import { useAuthUserOrNull } from '@frontegg/react-hooks';
+import { useAuthUserOrNull, useEntitlements } from '@frontegg/react-hooks';
 
 export interface AuthorizationProps {
   requiredRoles?: string[];
   requiredPermissions?: string[];
+  requiredEntitlements?: string[];
   render?: (isAuthorized: boolean) => React.ReactNode | null;
   children?: ReactNode;
 }
 
-export const AuthorizedContent: FC<AuthorizationProps> = (props) => {
+export const AuthorizedContent: FC<AuthorizationProps> = ({
+  requiredRoles,
+  requiredPermissions,
+  requiredEntitlements,
+  render,
+  children,
+}) => {
   let isAuthorized = true; // Initially
   const user = useAuthUserOrNull();
+  const entitlements = useEntitlements(requiredEntitlements || []);
 
   if (!user?.superUser) {
-    if (props.requiredPermissions) {
+    if (requiredPermissions) {
       isAuthorized = false; // Reset - we are going to check that the user has at least one matching permission
-      for (const permission of props.requiredPermissions) {
+      for (const permission of requiredPermissions) {
         if (user?.permissions?.find(({ key }) => key === permission)) {
           isAuthorized = true;
+          break;
         }
       }
     }
 
-    if (props.requiredRoles) {
+    if (requiredRoles) {
       isAuthorized = false; // Reset - we are going to check that the user has at least one matching role
-      for (const role of props.requiredRoles) {
+      for (const role of requiredRoles) {
         if (user?.roles?.find(({ key }) => key === role)) {
           isAuthorized = true;
+          break;
         }
       }
     }
+
+    if (isAuthorized && requiredEntitlements) {
```

Comment:
I think that this solution is not supporting using AuthorizedContent only with requiredEntitlements, cause you will not change isAuthorized to true in any way. correct?

Reply:
I don't see the issue. 
Let's say:
```json
{
  requiredRoles: undefined,
  requiredPermissions: undefined,
  requiredEntitlements: ['feature-1'],
}
```
Then in line `19` we change `isAuthorized` to `true`, we skip all conditions and react line `44` with `true` so we will enter the condition and convert the boolean value to `false` if the user is not entitled. 
Can you give an example?

Reply:
I missed that the initial value of isAuthorized is true. :)